operator/watchers/node_taint.go

```diff
@@ -127,74 +145,23 @@ func ciliumPodsWatcher(wg *sync.WaitGroup, clientset k8sClient.Clientset, stopCh
 		0,
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
-				key, _ := queueKeyFunc(obj)
-				ciliumQueue.Add(key)
+				ciliumPodHandler(obj, queue, logger)
 			},
 			UpdateFunc: func(_, newObj interface{}) {
-				key, _ := queueKeyFunc(newObj)
-				ciliumQueue.Add(key)
+				ciliumPodHandler(newObj, queue, logger)
 			},
```

Comment:
I just realized we don't specify `DeleteFunc`; that's not good :-)

Reply:
yes and no, I saw that too, we are not tainting the node when the pod is deleted, only when is scheduled but not running. 
Theoretically, someone could change nodeSelector and Cilium would never be run there again :shrug: 
What I mean is that even if we add delete event, checkAndMarkNode thing will ignore event.